core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -60,13 +71,19 @@ private[spark] object JsonProtocol {
   private val mapper = new ObjectMapper().registerModule(DefaultScalaModule)
     .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
+  val defaultOptions: JsonProtocolOptions = new JsonProtocolOptions(new SparkConf(false))
+
   /** ------------------------------------------------- *
    * JSON serialization methods for SparkListenerEvents |
    * -------------------------------------------------- */
 
   def sparkEventToJsonString(event: SparkListenerEvent): String = {
```

Comment:
Method using default configurations is retained for the purposes of avoiding changes in a bunch of test code across several suites.